backend/shop/views.py

```diff
@@ -274,3 +279,29 @@ def role_create(request):
 def permission_user_get(request):
     permissions = list(request.user.get_all_permissions())
     return Response({"permissions": permissions}, status=status.HTTP_200_OK)
+
+
+@api_view(["GET"])
+@login_required()
+def product_get(request, product_id=None):
+    if not can_view_products(request.user):
+        raise PermissionDenied("You don't have permission to view products.")
+    if product_id:
+        product = Product.objects.get(id=product_id)
```

Comment:
so everyone can see all the product?
what happens if you request invalid or deleted id?

Reply:
No...
In the lines above:
```
if not can_view_products(request.user):
    raise PermissionDenied("You don't have permission to view products.")
```
is condition permission.